### PR TITLE
Support context with Project[.BuildType][+TargetType] symantic

### PR DESCRIPTION
**Pull request by soumeh01**
Addressed [requirement](https://github.com/Open-CMSIS-Pack/cbuild/issues/25) 

Since `Project`, `TargetType` and `BuildType` are optional in the context. The change includes support of context with
`[Project][.BuildType][+TargetType]` symantic. The valid contexts would be
- Project
- +TargetType
- .BuildType
- Project+TargetType
- Project.BuildType
- .BuildType+TargetType
- +TargetType.BuildType
- Project.BuildType+TargetType
- Project+TargetType.BuildType 

The change includes:
- Support for contexts with/without Project, BuildType & TargetType
- From all contexts, the projects with the filtered contexts shall be converted and built sequentially.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Available Commands:
 
 Flags:
   -c, --clean              Remove intermediate and output directories
-      --context string     Input context name e.g. project.buildType+targetType
+      --context string     Input context name e.g. project[.buildType][+targetType]
```
Comment (brondani) [inline README.md:29]: From [this requirement](https://github.com/Open-CMSIS-Pack/cbuild/issues/25#issue-1468149063) I believe the `[project]` part of the context string should be optional as well. In other words the `[project]` may be empty and the selected contexts should be handled accordingly.

Comment (jkrech) [reply README.md:29]: I agree, but only once we support to orchestrate multiple project build.

Comment (brondani) [reply README.md:29]: Isn't multiple project build support being introduced in this PR?
From its [description](https://github.com/Open-CMSIS-Pack/cbuild/pull/35#issue-1566294565):
_From all contexts the projects with the filtered contexts shall be converted and built sequentially._

Comment (soumeh01) [reply README.md:29]: @brondani  Yes, your observation is right. It has been introduced in this PR. With this PR, project contexts are built one after the other. Currently, the order of execution is not controlled.
I believe we should make the change related to ` [project]`.  It should be doable now.

Comment (jkrech) [reply README.md:29]: Apologies.

Comment (soumeh01) [reply README.md]: Addressed here
[Support optional project name in [project][.buildType][+targetType]](https://github.com/Open-CMSIS-Pack/cbuild/pull/35/commits/1c4ab2cea399cdbef7ba0a42487020c39874e5d7)

**Review by brondani — DISMISSED** [review]

**Review by brondani — APPROVED** [review]

**Pull request merged**